aip/general/0127.md

```diff
@@ -73,18 +73,24 @@ message CreateBookRequest {
     should be populated in the request proto. When extracting a [resource
     name][aip-122], the variable **must** include the entire resource name, not
     just the ID component.
+  - URIs **may** use nested fields for their variable names. (Additionally,
+    AIP-134 mandates this for `Update` requests.)
   - URIs **must** use the `*` character to represent ID components, which
     matches all URI-safe characters except for `/`. URIs **may** use `**` as
     the final segment of a URI if matching `/` is required.
-- The `body` key defines which field in the request will be sent as the HTTP
-  body. If the body is `*`, then this indicates that the request object itself
-  is the HTTP body. The request body is encoded as JSON as defined by protocol
-  buffers' canonical [JSON encoding][].
+- The `body` key defines which single top-level field in the request will be
+  sent as the HTTP body. If the body is `*`, then this indicates that the
+  request object itself is the HTTP body. The request body is encoded as JSON
+  as defined by protocol buffers' canonical [JSON encoding][].
   - RPCs **must not** define a `body` at all for RPCs that use the `GET` or
     `DELETE` HTTP verbs.
   - RPCs **must** use the prescribed `body` for Create ([AIP-133][]) and Update
     ([AIP-134][]) requests.
   - RPCs **should** use the prescribed `body` for custom methods ([AIP-136][]).
+  - The `body` **must not** contain a nested field (or use the `.` character),
+  - The `body` **must not** be the same as a URI parameter.
+  - The `body` **must not** be a `repeated` field.
+  - The `body` **should** only be a primitive field on `GET` requests.
```

Comment:
Should this aip suggest having `body` for GET requests at all, given the the unclear semantics?
> A payload within a GET request message has no defined semantics; sending a payload body on a GET request might cause some existing implementations to reject the request.

https://tools.ietf.org/html/rfc7231#page-24

Reply:
Good point. We say elsewhere that `GET` should have no body.